src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/FunctionsTests.cs

```diff
@@ -659,6 +662,64 @@ private void TestUdf()
                 (arg) => new Dictionary<string, string[]> { { arg, new[] { arg } } });
         }
 
+        [Fact]
+        public void CatalogFunctions()
+        {
+            var spark = SparkSession.Builder().GetOrCreate();
+            var catalog = spark.Catalog();
+
+            Assert.IsType<DataFrame>(catalog.ListDatabases());
+            Assert.IsType<DataFrame>(catalog.ListFunctions());
+            Assert.IsType<DataFrame>(catalog.ListFunctions("default"));
+            
+            var table = catalog.CreateTable("users", TestEnvironment.ResourceDirectory + "users.parquet");
+            Assert.IsType<DataFrame>(table);
+
+            Assert.IsType<string>(catalog.CurrentDatabase());
+            Assert.IsType<bool>(catalog.DatabaseExists("default"));
+
+            Assert.IsType<bool>(catalog.DropGlobalTempView("no-view"));
+            Assert.IsType<bool>(catalog.DropTempView("no-view"));
+            Assert.IsType<bool>(catalog.FunctionExists("default", "functionname"));
+            Assert.IsType<bool>(catalog.FunctionExists("functionname"));
+            Assert.IsType<Database>(catalog.GetDatabase("default"));
+            Assert.IsType<Function>(catalog.GetFunction("abs"));
+            Assert.IsType<Function>( catalog.GetFunction(null, "abs"));
+            Assert.IsType<Table>(catalog.GetTable("users"));
+            Assert.IsType<Table>(catalog.GetTable("default", "users"));
+            Assert.IsType<bool>(catalog.IsCached("users"));
+            Assert.IsType<DataFrame>(catalog.ListColumns("users"));
+            Assert.IsType<DataFrame>(catalog.ListColumns("default", "users"));
+            Assert.IsType<DataFrame>(catalog.ListDatabases());
+            Assert.IsType<DataFrame>(catalog.ListFunctions());
+            Assert.IsType<DataFrame>(catalog.ListFunctions("default"));
+            Assert.IsType<DataFrame>(catalog.ListTables());
+            Assert.IsType<DataFrame>(catalog.ListTables("default"));
+            
+            catalog.RefreshByPath("/");
+            catalog.RefreshTable("users");
+            catalog.SetCurrentDatabase("default");
+            catalog.CacheTable("users");
+            catalog.UncacheTable("users");
+            catalog.ClearCache();
+
+            Assert.IsType<bool>(catalog.TableExists("users"));
+            Assert.IsType<bool>(catalog.TableExists("default", "users"));
+
+            //if the test doesn't get to drop the paritioned table then spark-warehouse is leaving files
```

Comment:
Can't you just set this to TestEnvironment.ResourceDirectory, so that it gets automatically cleaned up?

Reply:
If the test ends before the delete is called a folder is left in the spark-warehouse, when the test runs again spark won’t overwrite the folder so the test fails on the create table call, calling delete table first doesn’t work.

It isn’t strictly necessary if the test always gets past the table delete otherwise it would be a flaky test that is hard to debug. 

Another option would be to create a random name for the table for each test run.

Reply:
Can you use [TemporaryDirectory](https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark.E2ETest/Utils/TemporaryDirectory.cs) ?

Reply:
hmmm, maybe I can set spark.sql.warehouse.dir to TemporaryDirectory, i'll give it a go :)

Reply:
setting "spark.sql.warehouse.dir" to "TemporaryDirectory/spark-warehouse" worked - I tried TestEnvironment.ResourceDirectory but it isn't cleaned out before a test run.